tests/components/watergate/test_init.py

```diff
@@ -0,0 +1,94 @@
+"""Tests for the Watergate integration init module."""
+
+from unittest.mock import AsyncMock, patch
+
+from homeassistant.components.watergate import get_webhook_handler
+from homeassistant.components.watergate.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from tests.common import ANY
+
+
+async def test_async_setup_entry(hass: HomeAssistant, mock_entry) -> None:
+    """Test setting up the Watergate integration."""
+    hass.config.components.add("network")
+    hass.config.internal_url = "http://hassio.local"
+    mock_entry.add_to_hass(hass)
+
+    with (
+        patch(
+            "homeassistant.components.watergate.WatergateLocalApiClient"
+        ) as mock_client,
+        patch(
+            "homeassistant.components.watergate.WatergateDataCoordinator"
+        ) as mock_coordinator,
```

Comment:
normally we don't patch the whole coordinator, but only your library inside the coordinator